pkg/objectstate/rte/machineconfigpool.go

```diff
@@ -108,12 +108,12 @@ func (obj machineConfigPoolFinder) FindState(mf Manifests, tree nodegroupv1.Tree
 		}
 
 		gdm := GeneratedDesiredManifest{
-			ClusterPlatform:       obj.em.plat,
-			MachineConfigPool:     mcp.DeepCopy(),
-			NodeGroup:             tree.NodeGroup.DeepCopy(),
-			DaemonSet:             desiredDaemonSet,
-			RTEConfigHash:         rteComputedConfigHash,
-			IsCustomPolicyEnabled: annotations.IsCustomPolicyEnabled(tree.NodeGroup.Annotations),
```

Comment:
wrong change: this struct is unfortunately part of the public API `pkg/objectstate/...` so we should strive to maximize the backward compat